plugins/urls/__init__.py

```diff
@@ -1,168 +1,209 @@
-
-import system.plugin as plugin
-
 from system.protocols.generic.channel import Channel
-from system.protocols.generic.user import User
-
-from system.storage.formats import YAML, DBAPI
+from system.storage.formats import Formats
+from system.plugins.plugin import PluginObject
 from system.storage.manager import StorageManager
+from plugins.urls.constants import PREFIX_TRANSLATIONS
+from plugins.urls.events import URLsPluginLoaded
+from plugins.urls.handlers.website import WebsiteHandler
+from plugins.urls.matching import extract_urls
+from plugins.urls.shorteners.tinyurl import TinyURLShortener
+from plugins.urls.url import URL
+from utils.misc import str_to_regex_flags
 
-from .catcher import Catcher
+__author__ = 'Gareth Coles'
 
-from system.translations import Translations
-_ = Translations().get()
-__ = Translations().get_m()
 
+HTTP_S_REGEX = re.compile("http|https", flags=str_to_regex_flags("iu"))
 
-class URLsPlugin(plugin.PluginObject):
-    """
-    URLs plugin object
-    """
 
-    catcher = None
-    channels = None
+class URLsPlugin(PluginObject):
+    # Managers
     commands = None
-    config = None
     events = None
-    shortened = None
     storage = None
 
-    blacklist = []
-    handlers = {}
-    shorteners = {}
-    spoofing = {}
+    channels = None
+    config = None
+    shortened = None
 
-    content_types = ["text/html", "text/webviewhtml", "message/rfc822",
-                     "text/x-server-parsed-html", "application/xhtml+xml"]
+    shorteners = None
+    handlers = None
 
     def setup(self):
-        """
-        Called when the plugin is loaded. Performs initial setup.
-        """
-
-        self.logger.trace(_("Entered setup method."))
         self.storage = StorageManager()
+        self.events = EventManager()
+        self.commands = CommandManager()
+
+        self.shorteners = {}
+        self.handlers = defaultdict(list)
+
+        # Load up the configuration
+
         try:
-            self.config = self.storage.get_file(self, "config", YAML,
-                                                "plugins/urls.yml")
+            self.config = self.storage.get_file(
+                self, "config", Formats.YAML, "plugins/urls.yml"
+            )
         except Exception:
-            self.logger.exception(_("Error loading configuration!"))
-        else:
-            if not self.config.exists:
-                self.logger.warn(_("Unable to find config/plugins/urls.yml"))
-            else:
-                self.content_types = self.config["content_types"]
-                self.spoofing = self.config["spoofing"]
-
-        self.logger.debug(_("Spoofing: %s") % self.spoofing)
+            self.logger.exception("Error loading configuration")
+            return self._disable_self()
+
+        if not self.config.exists:
+            self.logger.error("Unable to load configuration: File not found")
+            self.logger.error("Did you fill out `config/plugins/urls.yml`?")
```

Comment:
This should be one logger call. This would result in, for example, two separate emails.